### ISSUE TITLE
App Crash on Library Scan

### ISSUE THREAD

**Issue by Artiume** [jellyfin-archive/jellyfin-roku-legacy#302]
When a Library is updated and you're watching a stream, when you exit the stream and go back to the Home Page. After the Home Page updates showing the new library content, the app will exit out to the Roku Main Menu, causing you to reopen the Jellyfin App.

**Comment by cewert** [comment]
This just happened to me and I was able to reproduce it. I sat on the home page scrolling around in some of the "Latest In" rows when I got the error below. @neilsb is there a way we can do this without the GetParent() call?

```
013:*   if m.top.GetParent().content.imageWidth <> invalid
014:      imageWidth = m.top.GetParent().content.imageWidth
015:    end if
016:  
017:    itemPoster.width = imageWidth
Source Digest(s): 
pkg: dev 1.3.5 ef955eade0db73b4d4a08c52d58ce3ae Jellyfin

'Dot' Operator attempted with invalid BrightScript Component or interface reference. (runtime error &hec) in pkg:/components/home/HomeItem.brs(13)```

**Comment by neilsb** [comment]
OK - so I _think_ this must be due to one of the Roku OS updates.  What we noticed happening (#299) was that sometimes the PlaceHolder images in the Item Grid were blank, rather than blue.   What I tracked it down to was Roku re-using the ContentNode object, when it went out of view.  So rather than destroying the old object and creating a new one, it just updated the content in the one that was out of view and then re-displayed it.   The net result of this was that the `init()` function was not run again, which is where the default properties were set (to display the blue background until the poster image was loaded).

So this makes sense from a performance point of view.

I think the same is happening here, although in this case what we're seeing is that the Node must be removed from parent when it goes out view, then the new content applied (which is why GetParent is null at this point - but not when the item is originally created) and then re-displayed.

PR #320 sets the `usePoster` and `imageWidth` against each item as it adds it to the row, meaning the item itself no longer needs to get that info from teh Parent, so should work even when it's being recycled.   There is an assumption here that the ContentNode will only be recycled within the HomeRow in which is was created, (otherwise these properties may be incorrect) but that _should be_ how Roku does it
